### PR TITLE
WE-467: GTM script update

### PR DESCRIPTION
**Pull request by maxharrisnet**
### What Changed

- Adds new Google Tag Manager implementation

### How To Test or Verify

- Verify code in the inspector
- Use Tag Assistant to verify that tags are firing: https://tagassistant.google.com/

### PR Checklist

Below are some checklists to follow for the correct procedure in different circumstance. The first list ("All PRs Checklist") should be followed for ALL PRs. The next 2 are additive to this list depending on what type of PR you are using.

For example: If you are submitting a content change to one of the support documents, your checklist would include the:

- "All PRs Checklist"
- AND the "Content Changes Checklist

If you are submitting a feature addition, enhancement, or bug fix, your checklist would include the:

- "All PRs Checklist"
- AND the "Development Changes Checklist"

#### All PRs Checklist

- [ ] Give your pull request a meaningful name.
- [ ] Use lowercase filenames.
- [ ] Apply at least one team label according to which team is the content expert (ie. `team-FE` or `team-SAZ`)
- [ ] Pull request approval from the FE team or content experts (see label applied above) that isn't the content creator.

#### Content Changes Checklist

- [ ] Check that your article looks correct in the preview here or in a [Netlify deploy preview](https://app.netlify.com/sites/support-docs/deploys).
- [ ] Check the links in your article.
- [ ] Check the images in your article (if there are any)
- [ ] Check to make sure you are using markdown appropriately as outlined in `examples/article.md` in the root of the project directory and on the momentum doc's [preface article](https://support.sparkpost.com/momentum/4/4-preface)
- [ ] Check to make sure the [Copy and Tone Guidelines are followed](https://docs.google.com/document/d/1dej9J7N9M8lcbJXnT9kxyNB_EFBPHIBLZBbHaxRWqhQ/edit).

#### Development Changes Checklist (some checks are automatic github actions and will not be listed here. ie. "all tests pass")

- [ ] The appropriate tests are created in `cypress/` directory in the root of the project
- [ ] The lighthouse score is passing according to the [FE Support Docs' Service Outline](https://sparkpost.atlassian.net/wiki/spaces/ENG/pages/1238728726/FE+Support+Docs) SLI/SLOs

### CODE REVIEW

File: components/site/gtm.tsx [hunk 1]
```diff
@@ -12,8 +12,8 @@ const GTM = (): JSX.Element => {
             (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
             new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
             j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
-            '//www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
-            })(window,document,'script','dataLayer', '${process.env.NEXT_PUBLIC_GTM_ID}');
+            'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
```
Comment (EvanYoung-SP) [inline components/site/gtm.tsx:15]: Do we need to remove the https so that it works fine on local machines or is this necessary for it to work?

**Review by EvanYoung-SP — CHANGES_REQUESTED** [review]
Just one question we should discuss. :-)

File: components/site/gtm.tsx (hunk 1 above)
Comment (maxharrisnet) [reply components/site/gtm.tsx]: Yes, one moment

**Review by EvanYoung-SP — APPROVED** [review]
LGTM!

**Pull request closed without merging**